src/ApiProvider.php

```diff
@@ -45,6 +53,10 @@ public function __construct(string $token)
      */
     public function getClient(): ClientInterface
     {
+        if (!is_null($this->rateLimiter)) {
+            ($this->rateLimiter)();
```

Comment:
I would use `call_user_func($this->rateLimiter)` here in case if user passes the callable as `[$object, 'method']` or function name/static method name as `string`.